src/cmd/services/m3coordinator/ingest/write.go

```diff
@@ -125,12 +128,28 @@ func NewDownsamplerAndWriter(
 	instrumentOpts instrument.Options,
 ) DownsamplerAndWriter {
 	scope := instrumentOpts.MetricsScope().SubScope("downsampler")
+
+	sourceTags := map[ts.SourceType]string{
+		ts.SourceTypePrometheus:  "prometheus",
```

Comment:
nit: maybe `SourceType` could have `String()`method so there would be no need for this ad-hoc name mapping here?

Reply:
This was actually the approach that I tried initially, but then decided to abandon it. I wanted to avoid coupling of enum names with metric labels because:
1. Differences in formatting, e.g. `SourceTypeOpenMetrics` vs `open-metrics`.
2. If someone renames an internal enum member, metric label should still remain the same (who knows, maybe there is an alert on it).

Reply:
You could define a method on `ts.SourceType` "asMetricLabel" and maybe add a comment to be careful if changing.

Reply:
But I would still miss a way to enumerate the values as a limitation of golang enums... Except for https://stackoverflow.com/a/69396558/17485516 which I don't want to use.